internal/commands/vaultsecrets/integrations/create_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-openapi/runtime/client"
+	"github.com/stretchr/testify/require"
+
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	preview_models "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/models"
+	mock_preview_secret_service "github.com/hashicorp/hcp/internal/pkg/api/mocks/github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+func TestNewCmdCreate(t *testing.T) {
+	t.Parallel()
+
+	testProfile := func(t *testing.T) *profile.Profile {
+		tp := profile.TestProfile(t).SetOrgID("123").SetProjectID("456")
+		return tp
+	}
+
+	cases := []struct {
+		Name    string
+		Args    []string
+		Profile func(t *testing.T) *profile.Profile
+		Error   string
+		Expect  *CreateOpts
+	}{
+		{
+			Name:    "Good",
+			Profile: testProfile,
+			Args:    []string{"sample-integration", "--config-file", "path/to/file"},
+			Expect: &CreateOpts{
+				IntegrationName: "sample-integration",
+				ConfigFilePath:  "path/to/file",
+			},
+		},
+		{
+			Name:    "Failed: No secret name arg specified",
+			Profile: testProfile,
+			Args:    []string{"--config-file", "path/to/file"},
+			Error:   "ERROR: accepts 1 arg(s), received 0",
+		},
+		{
+			Name:    "Failed: No config file flag specified",
+			Profile: testProfile,
+			Args:    []string{"sample-integration"},
+			Error:   "ERROR: missing required flag: --config-file=CONFIG_FILE",
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			t.Parallel()
+			r := require.New(t)
+
+			// Create a context.
+			io := iostreams.Test()
+			ctx := &cmd.Context{
+				IO:          io,
+				Profile:     c.Profile(t),
+				Output:      format.New(io),
+				HCP:         &client.Runtime{},
+				ShutdownCtx: context.Background(),
+			}
+
+			var gotOpts *CreateOpts
+			createCmd := NewCmdCreate(ctx, func(o *CreateOpts) error {
+				gotOpts = o
+				return nil
+			})
+			createCmd.SetIO(io)
+
+			code := createCmd.Run(c.Args)
+			if c.Error != "" {
+				r.NotZero(code)
+				r.Contains(io.Error.String(), c.Error)
+				return
+			}
+
+			r.Zero(code, io.Error.String())
+			r.NotNil(gotOpts)
+			r.Equal(c.Expect.IntegrationName, gotOpts.IntegrationName)
+			r.Equal(c.Expect.ConfigFilePath, gotOpts.ConfigFilePath)
+		})
+	}
+}
+
+func TestCreateRun(t *testing.T) {
```

Comment:
this looks like a twilio specific test, so maybe a rename to `TestCreateTwilioRun` might make sense?

Reply:
Hmmm, that's a good point. I'll include 2 other providers in the test cases.

Reply:
thanks!!